add com.gitlab.davem.ClamTk

refi64 opened the pull request:
Uses the new [flatpak-cpan-generator](https://github.com/flatpak/flatpak-builder-tools/pull/68).

LeandroStanger commented:
bot, build com.gitlab.davem.ClamTk

AsciiWolf commented on com.gitlab.davem.ClamTk.appdata.xml:
```diff
@@ -0,0 +1,24 @@
+  <url type="homepage">https://gitlab.com/dave_m/clamtk-gtk3</url>
+  <description>
+    <p>ClamTk is a frontend for ClamAV (Clam Antivirus). It is intended to be an easy to use, light-weight, on-demand scanner for Linux systems. It has been ported to Fedora, Debian, RedHat, openSUSE, ALT Linux, Ubuntu, CentOS, Gentoo, Archlinux, Mandriva, PCLinuxOS, Frugalware, FreeBSD, and others.</p>
+    <p>Although its earliest incarnations date to 2003, ClamTk was first uploaded for distribution in 2004 to a rootshell.be account and finally to Sourceforge.net in 2005. At the end of 2013, it was moved to a Google Code page (then to github), then gitlab and Bitbucket. It's now 2019 and for some reason, it's still going. In fact, February 2019 marks 15 years of activity (of being publically av ...
```
I am not sure if this information is somehow useful for the user. Maybe some details about the ClamAV antivirus itself could be more useful. :-)

refi64 commented:
I can't seem to find any official screenshots that aren't insanely out-of-date, is it considered okay if I take one myself and place it in this repo?

bilelmoussaoui commented:
> I can't seem to find any official screenshots that aren't insanely out-of-date, is it considered okay if I take one myself and place it in this repo?

Yes please do. A bunch of apps keep their screenshots downstream.
Were the maintainers contacted? If not, it would be a good opportunity to ask if they want to provide screenshots themselves.

bilelmoussaoui commented:
Also, it would be nice to upstream the appdata & the rest of the patches.

bilelmoussaoui commented on com.gitlab.davem.ClamTk.yaml:
```diff
@@ -0,0 +1,121 @@
+app-id: com.gitlab.davem.ClamTk
+runtime: org.freedesktop.Platform
+runtime-version: '18.08'
```
Have you tried building it using 19.08?

barthalion replied:
Yes please, while 18.08 is still supported, we definitely don't want more apps on "old" runtimes.

bilelmoussaoui commented on com.gitlab.davem.ClamTk.yaml:
```diff
@@ -0,0 +1,121 @@
+  - '--share=network'
+  - '--socket=x11'
+  - '--socket=wayland'
+  - '--filesystem=xdg-run/dconf'
```
What the app do with dconf access? if it's only for the app's settings, then those could be removed if migrated to 19.08.

refi64 replied:
It's used for the GTK open folder dialog to have favorites, if that's not wanted though I can drop it.

bilelmoussaoui replied:
Is the favorites are that important in such case? if not, can you drop the dconf access? Thanks

refi64 commented:
Okay so talking with upstream now, added a custom screenshot (the path is to where the repo will be created, not the current forked path). All the patches but 2 (the Flatpak paths patch, because the official upstream directions seem to imply that manual path changes are expected, and the dropping of auto update) were already submitted to and merged upstream in 6.02, which is now being used here.

Also I switched to 19.08, no idea why I used 18.08 before other than force of habit.

bilelmoussaoui commented:
bot, build com.gitlab.davem.ClamTk

gasinvein commented on com.gitlab.davem.ClamTk.yaml:
```diff
@@ -0,0 +1,117 @@
+rename-icon: clamtk
+copy-icon: true
+finish-args:
+  - '--filesystem=host'
```
Does this app need to write something outside xdg dirs? If not, maybe set `host:ro` instead?

refi64 replied:
True, fixed.

refi64 replied:
Actually that was a bad idea, because ClamTk can quarantine scanned files. So yeah, it does need to write (or more particularly, delete) outside the XDG dirs.

refi64 commented:
Okay so I retract the statement about the screenshot URL pointing to the future repo, because that made the validation fail. I'll change it after the repo.

refi64 commented:
bot, build com.gitlab.davem.ClamTk

bilelmoussaoui commented on com.gitlab.davem.ClamTk.yaml:
```diff
@@ -0,0 +1,117 @@
+        cp -r lib $sitelib/ClamTk
+      - 'install -Dm 644 clamtk.desktop /app/share/applications/com.gitlab.davem.ClamTk.desktop'
+      - 'install -Dm 644 images/clamtk.png /app/share/icons/hicolor/128x128/apps/clamtk.png'
+      - 'install -Dm 644 com.gitlab.davem.ClamTk.appdata.xml /app/share/metainfo/com.gitlab.davem.ClamTk.appdata.xml'
```
```suggestion
      - 'install -Dm 644 com.gitlab.davem.ClamTk.metainfo.xml /app/share/metainfo/com.gitlab.davem.ClamTk.metainfo.xml'
```
Newer appdata files shouldn't use .appdata any more. See https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

bilelmoussaoui commented on com.gitlab.davem.ClamTk.appdata.xml:
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
```
```suggestion
<component type="desktop-application">
```

bilelmoussaoui commented:
bot, build  com.gitlab.davem.ClamTk

bilelmoussaoui commented:
bot, build com.gitlab.davem.ClamTk

bilelmoussaoui commented:
bot, build com.gitlab.davem.ClamTk

bilelmoussaoui commented:
Is anyone from upstream interested in maintaining/co-maintaining this? Also, the app is missing a bunch of icons here with the default Adwaita icon theme.

refi64 commented:
> Is anyone from upstream interested in maintaining/co-maintaining this?

No response yet (FWIW it's a single dev so it sometimes takes a bit).

> Also, the app is missing a bunch of icons here with the default Adwaita icon theme.

Which icons? The 6.02 update should have fixed those, and they look fine over here...

bilelmoussaoui commented:
> > Also, the app is missing a bunch of icons here with the default Adwaita icon theme.
> 
> Which icons? The 6.02 update should have fixed those, and they look fine over here...

![Screenshot from 2019-10-12 03-26-34](https://user-images.githubusercontent.com/7660997/66692586-2919f400-eca0-11e9-8685-d1ee152701c9.png)
![Screenshot from 2019-10-12 03-26-05](https://user-images.githubusercontent.com/7660997/66692587-29b28a80-eca0-11e9-8099-f7646de87e37.png)

ramcq commented:
bot, build com.gitlab.davem.ClamTk

refi64 commented:
Okay so:

- Icons should all work.
- Had to drop the scheduler for now because that requires editing the crontab, and given the current upstream implementation it will keep running even if you uninstall ClamTk. IMO for most usage this isn't terrible, but I might take a stab at having the crontab entry auto-remove itself if you drop the Flatpak.

LeandroStanger commented:
bot, build com.gitlab.davem.ClamTk

bilelmoussaoui commented:
> Okay so:
> 
>     * Icons should all work.

Perfect

>     * Had to drop the scheduler for now because that requires editing the crontab, and given the current upstream implementation it will keep running even if you uninstall ClamTk. IMO for most usage this isn't terrible, but I might take a stab at having the crontab entry auto-remove itself if you drop the Flatpak.

That could be added later

bilelmoussaoui commented:
/merge

bilelmoussaoui commented:
/merge

barthalion commented:
/merge
